### PR TITLE
Add retry for getDocuments

### PR DESCRIPTION
**Pull request by HashEngineering**
* Add support for handling InvalidContractIdError, when contract should exist

### CODE REVIEW

File: src/main/kotlin/org/dashevo/dapiclient/model/GrpcExceptionInfo.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        val cursor = trailers.findAnyOf(listOf("errors="))
+        val end = trailers.findLastAnyOf(listOf(")"))
+        val endpos = end?.first ?: trailers.length
+        if (cursor != null) {
+            val errorString = trailers.substring(cursor.first+ 7, endpos)
+            val json = "{errors: $errorString }"
+            val moshi = Moshi.Builder().build()
+            val map = Types.newParameterizedType(MutableMap::class.java, String::class.java, Any::class.java)
+            val adapter: JsonAdapter<Map<String, Any>> = moshi.adapter<Map<String, Any>>(map).lenient()
+            val errorMap = adapter.fromJson(json)
```
Comment (tomasz-ludek) [inline src/main/kotlin/org/dashevo/dapiclient/model/GrpcExceptionInfo.kt:24]: Here you can use the fallowing code to remove trailing commas if needed:
```
    private fun removeTrailingCommas(json: String): String {
        return json.replace("\r\n", "\n")
            .replace(Regex(",[\\s]*]"), "]")
            .replace(Regex(",\\s*\\}"), "}")
    }
```

**Review by tomasz-ludek — APPROVED** [review]
One comment regarding our discussion on Slack

File: src/main/kotlin/org/dashevo/dapiclient/model/GrpcExceptionInfo.kt (hunk 1 above)
Comment (HashEngineering) [reply src/main/kotlin/org/dashevo/dapiclient/model/GrpcExceptionInfo.kt:24]: great.  I will need that in android-dpp, which is where all the trailing commas are.  That will be a different PR!

File: build.gradle [hunk 2]
```diff
@@ -65,8 +65,9 @@ dependencies {
     implementation "org.dashj:dashj-core:$dashj_version"
     implementation 'com.squareup.retrofit2:retrofit:2.4.0'
     implementation 'com.squareup.retrofit2:converter-gson:2.4.0'
-    implementation 'com.google.code.gson:gson:2.8.5'
     implementation 'com.squareup.okhttp3:logging-interceptor:3.8.0'
+    implementation 'com.squareup.moshi:moshi:1.11.0'
+    implementation 'com.squareup.moshi:moshi-kotlin:1.11.0'
```
[the remark this answers is not in the document]
Comment (HashEngineering) [reply build.gradle:70]: Thank you!

**Comment by HashEngineering** [comment]
This was merged manually.

**Pull request closed without merging**